source/testing/testing-components.md

```diff
@@ -180,3 +180,94 @@ test('should trigger external action on form submit', function(assert) {
   this.$('input').click();
 });
 ```
+### Stubbing Services
+
+In cases where components have dependencies on Ember services, it is possible to stub these
+dependencies for integration tests. Stub Ember services by using the registry
+object to register your stub service in place of the default.
+
+Imagine you have the following component that uses a location service to display the city
+and country of your current location:
+
+> You can follow along by generating your own component with `ember generate
+> component location-indicator`.
+
+```app/components/location-indicator.js
+export default Ember.Component.extend({
+  locationService: Ember.inject.service('location-service'),
+
+  //when the coordinates change, call the location service to evaluate what the city and country would be
+  city: Ember.computed('locationService.currentLocation', function () {
+    return this.get('locationService').getCurrentCity();
+  }),
+
+  country: Ember.computed('locationService.currentLocation', function () {
+    return this.get('locationService').getCurrentCountry();
+  })
+});
+```
+
+```app/templates/components/location-indicator.hbs
+You currently are located in {{city}}, {{country}}
+```
+To stub the location service in your test, create a local stub object that extends
+`Ember.Service`, and register the stub as the service your tests need in the
+beforeEach function.  In this case we initially force location to New York.
+
+
+```tests/integration/components/location-indicator-test.js
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+//Stub location service
+let locationStub = Ember.Service.extend({
+  city: 'New York',
+  country: 'USA',
+  currentLocation: {
+    x: 1234,
+    y: 5678
+  }
+  getCurrentCity() {
+    return this.get('city');
+  },
+  getCurrentCountry() {
+    return this.get('country');
+  }
+});
+
+moduleForComponent('location-indicator', 'Integration | Component | location indicator', {
+  integration: true,
+
+  beforeEach: function () {
+    this.registry.register('service:location-service', locationStub);
```

Comment:
The code being added in https://github.com/switchfly/ember-test-helpers/pull/105